### PR TITLE
Source Exchange Rates - Refactor for API Layer

### PR DESCRIPTION
**Pull request by jonathanneo**
## What
The [Rates API](https://exchangeratesapi.io/) used for the [Airbyte demo for Python CDK (HTTP) connectors](https://docs.airbyte.com/connector-development/tutorials/cdk-tutorial-python-http/getting-started) has been deprecated. 

The Rates API is now being served by API Layer. Here is the [new API documentation](https://apilayer.com/marketplace/exchangerates_data-api). 

This PR resolves the following issues: 
- [#14779 ](https://github.com/airbytehq/airbyte/issues/14779)
- [#15524 ](https://github.com/airbytehq/airbyte/issues/15524)

## How
1. Change `url_base` to the new url: `https://api.apilayer.com/exchangerates_data/`
2. Change `spec.yaml` to use `api_key` instead of `access_key` so that it aligns with the new API terminology
3. Add `request_headers()` to pass a header with `apikey` to the endpoint 

## Recommended reading order
1. `source_exchange_rates/spec.yaml` 
2. `source_exchange_rates/source.py`

## 🚨 User Impact 🚨
- The spec now uses `api_key` instead of `access_key` so that it aligns with the new API terminology.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

No unit tests were created by the original airbyte author. 

</details>

<details><summary><strong>Integration</strong></summary>

Running integration tests locally from the root of the repo using the command `./gradlew :airbyte-integrations:connectors:source-exchange-rates:integrationTest` results in the following error: 

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/jonathanneo/code/jonathanneo/airbyte/build.gradle' line: 198

* What went wrong:
A problem occurred evaluating root project 'airbyte'.
> java.lang.UnsupportedClassVersionError: com/bmuschko/gradle/docker/DockerRemoteApiPlugin has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

</details>

<details><summary><strong>Acceptance</strong></summary>

Running `docker run --rm -v $(pwd)/integration_tests:/integration_tests airbyte/source-exchange-rates:dev read --config /integration_tests/config.json --catalog /integration_tests/configured_catalog.json` resulted in a successful sync: 

```
{"type": "RECORD", "record": {"stream": "exchange_rates", "data": {"success": true, "timestamp": 1661265424, "historical": true, "base": "USD", "date": "2022-08-23", "rates": {"AED": 3.673103, "AFN": 88.999668, "ALL": 117.874989, "AMD": 404.929874, "ANG": 1.802425, "AOA": 429.28555, "ARS": 136.884802, "AUD": 1.442294, "AWG": 1.8055, "AZN": 1.702342, "BAM": 1.970947, "BBD": 2.019288, "BDT": 95.032196, "BGN": 1.960328, "BHD": 0.376956, "BIF": 2046, "BMD": 1, "BND": 1.396195, "BOB": 6.910705, "BRL": 5.092601, "BSD": 1.000071, "BTC": 4.6440968e-05, "BTN": 79.83403, "BWP": 12.83827, "BYN": 2.524362, "BYR": 19600, "BZD": 2.015862, "CAD": 1.29701, "CDF": 2001.000153, "CHF": 0.963275, "CLF": 0.03355, "CLP": 925.749715, "CNY": 6.836402, "COP": 4379.9, "CRC": 646.377717, "CUC": 1, "CUP": 26.5, "CVE": 111.11615, "CZK": 24.703608, "DJF": 178.04561, "DKK": 7.45388, "DOP": 53.25008, "DZD": 141.204424, "EGP": 19.171398, "ERN": 15, "ETB": 52.890421, "EUR": 1.002205, "FJD": 2.20175, "FKP": 0.823605, "GBP": 0.84506, "GEL": 2.789737, "GGP": 0.823605, "GHS": 9.824967, "GIP": 0.823605, "GMD": 54.424996, "GNF": 8754.999711, "GTQ": 7.738252, "GYD": 209.232723, "HKD": 7.84732, "HNL": 24.550132, "HRK": 7.529299, "HTG": 128.512692, "HUF": 414.357027, "IDR": 14850.4, "ILS": 3.262375, "IMP": 0.823605, "INR": 79.783499, "IQD": 1460, "IRR": 42350.000177, "ISK": 140.429707, "JEP": 0.823605, "JMD": 150.949785, "JOD": 0.709028, "JPY": 136.409007, "KES": 119.849744, "KGS": 80.328902, "KHR": 4114.50145, "KMF": 493.94986, "KPW": 900.00035, "KRW": 1336.929896, "KWD": 0.30775, "KYD": 0.833392, "KZT": 466.909875, "LAK": 15279.999655, "LBP": 1505.652441, "LKR": 365.277775, "LRD": 153.64994, "LSL": 16.989678, "LTL": 2.95274, "LVL": 0.60489, "LYD": 4.920557, "MAD": 10.503503, "MDL": 19.351829, "MGA": 4116.000089, "MKD": 62.090614, "MMK": 2100.158916, "MNT": 3194.961563, "MOP": 8.083256, "MRO": 356.999828, "MUR": 44.149314, "MVR": 15.35044, "MWK": 1022.502229, "MXN": 20.01538, "MYR": 4.484996, "MZN": 63.82977, "NAD": 16.990023, "NGN": 421.210168, "NIO": 35.949729, "NOK": 9.73107, "NPR": 127.734166, "NZD": 1.606935, "OMR": 0.384958, "PAB": 1.000071, "PEN": 3.868042, "PGK": 3.514995, "PHP": 56.014963, "PKR": 216.695056, "PLN": 4.78277, "PYG": 6887.728881, "QAR": 3.641017, "RON": 4.894301, "RSD": 117.659966, "RUB": 60.005502, "RWF": 1031.5, "SAR": 3.755612, "SBD": 8.220417, "SCR": 12.740352, "SDG": 570.000198, "SEK": 10.613525, "SGD": 1.392245, "SHP": 1.377402, "SLL": 13994.999805, "SOS": 290.834995, "SRD": 24.4855, "STD": 20697.981008, "SVC": 8.751121, "SYP": 2512.53037, "SZL": 16.9905, "THB": 36.030103, "TJS": 10.231075, "TMT": 3.51, "TND": 3.1855, "TOP": 2.353903, "TRY": 18.112403, "TTD": 6.789105, "TWD": 30.172502, "TZS": 2332.227985, "UAH": 36.930457, "UGX": 3845.394173, "USD": 1, "UYU": 40.508702, "UZS": 10910.000175, "VND": 23435, "VUV": 116.056079, "WST": 2.66746, "XAF": 661.000433, "XAG": 0.052207, "XAU": 0.000572, "XCD": 2.70255, "XDR": 0.765642, "XOF": 654.501052, "XPF": 120.324979, "YER": 250.249645, "ZAR": 16.9472, "ZMK": 9001.198797, "ZMW": 16.169016, "ZWL": 321.999592}}, "emitted_at": 1661265436541}}
{"type": "STATE", "state": {"data": {"exchange_rates": {"date": "2022-08-23"}}}}
{"type": "LOG", "log": {"level": "INFO", "message": "Read 6 records from exchange_rates stream"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing exchange_rates"}}
{"type": "LOG", "log": {"level": "INFO", "message": "SourceExchangeRates runtimes:\nSyncing stream exchange_rates 0:00:03.158786"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Finished syncing SourceExchangeRates"}}
```

</details>

### CODE REVIEW

**Comment by jonathanneo** [comment]
Hi @natalyjazzviolin , the build has failed. What are the next steps?

**Comment by natalyjazzviolin** [comment]
@jonathanneo I'm debugging it, should have an update for you soon :)

**Comment by jonathanneo** [comment]
Hi @natalyjazzviolin , any updates?

**Comment by sajarin** [comment]
@jonathanneo the integration tests are breaking on our end since we haven't updated our credentials with a new api key from APILayer. Just made an issue requesting for one to be made so please stay tuned. Appreciate the contribution, hope to get this merged as soon as it's unblocked. 

Let me know if you have questions :)

**Comment by jonathanneo** [comment]
Awesome! Thanks both for testing the change :)

**Comment by jonathanneo** [comment]
Hi @sajarin just following up :)

**Review by sajarin — APPROVED** [review]

File: docs/integrations/sources/exchangeratesapi.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ If you have `free` subscription plan \(you may check it [here](https://manage.ex
 
 | Version | Date | Pull Request | Subject |
 |:--------| :--- | :--- | :--- |
+| 0.2.7   | 2022-08-23 | [15884](https://github.com/airbytehq/airbyte/pull/15884) | Added support for new API Layer endpoint |
```
Comment (marcosmarxm) [inline docs/integrations/sources/exchangeratesapi.md:49]: @sajarin the dockerfile version is not matching with the version in the document.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/exchangeratesapi.md (hunk 1 above)
Comment (sajarin) [reply docs/integrations/sources/exchangeratesapi.md]: Should be good to go now?

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**